dotCMS/src/main/java/com/dotcms/content/business/ContentletJsonAPIImpl.java

```diff
@@ -140,7 +145,25 @@ public ContentletJsonAPIImpl() {
      */
     public String toJson(final com.dotmarketing.portlets.contentlet.model.Contentlet contentlet)
             throws JsonProcessingException, DotDataException {
-        return objectMapper.get().writeValueAsString(toImmutable(contentlet));
+
+        final com.dotmarketing.portlets.contentlet.model.Contentlet copy = new com.dotmarketing.portlets.contentlet.model.Contentlet(contentlet);
+
+        final ContentType contentType = copy.getContentType();
+        final ImmutableSet.Builder<com.dotcms.contenttype.model.field.Field> builder = ImmutableSet.builder();
+        Set<Field> categoriesAndTagFields = builder.addAll(contentType.fields(TagField.class))
+                .addAll(contentType.fields(CategoryField.class)).build();
+
+        final Map<String, Object> map = copy.getMap();
+        //A small precaution that guarantees categories and tags will be stored as Lists.
+        categoriesAndTagFields.forEach(field -> {
+            final Object object = map.get(field.variable());
+            if(object instanceof String){
+                final String asString = (String)object;
+                map.put(field.variable(), Arrays.asList(asString.split("\\s*,\\s*")));
```

Comment:
I would call here something such as 

new StringAsListTransformer().transform(asString)

Reply:
no need to new-up another middleman object. Do we really have something like that?? I would have added that to a utility class and made it static accessible.